### ISSUE TITLE
[V6] Field Arrays Example with immutablejs doesn't work.

### ISSUE THREAD

**Issue by tdziezyk** [erikras/redux-form#1071]
I changed reducer. Instead of:

```js
import { Field, FieldArray, reduxForm } from 'redux-form'
```

I put:

```js
import { Field, FieldArray, reduxForm } from 'redux-form/immutable'
```

and example is displaying properly, but when I add club name and try type text into 'first name' field I get an error:

```
Uncaught Error: invalid keyPath
```

Is there any solution/work around for it?

**Comment by clayne11** [comment]
Can you post a reproduction repo?

**Comment by kkwiatkowski** [comment]
I got the very same issue, but in version 5.2.5.

As far as I can see, everything started with this commit 7b5cb0f64911dffd2ac80e254719ec75333cb8f4
and introduction of `fieldValue.js`.

`makeFieldValue` is assigning not enumerable property to field object, so when trying to make Immutable from such object, not-enumerable property is lost.
It does matter, because on store, there is no `_isFieldValue` property assigned to field object, and because of that `isFieldValue` is failing when used in [`src/getValuesFromState`](https://github.com/erikras/redux-form/blob/v5.2.5/src/getValuesFromState.js#L17).

This is not only related to Immutable.js, but also to any form of state serialization, for example JSON.stringify.
I wanted to make a pull request with easy fix to make this prop enumerable, but it comes with loads of tests changes.
I think that `makeFieldValue` and `isFieldValue` should be rewritten in some way to get rid of using not enumerable props, or adding any props at all.

I hope it helps.

**Comment by erikras** [comment]
@kkwiatkowski Anything in `v5` is unrelated to the original issue posted here.

@tdziezyk Are you remembering to use a special [immutable-enabled version of `combineReducers` as well as the immutable version of the `redux-form` reducer](https://github.com/erikras/redux-form/blob/v6/examples/immutable/src/index.js#L5-L6)?

**Comment by jccf091** [comment]
I got the same issue in v6.0.0-rc.1.
@erikras I did use the combineReducers from 'redux-immutable' and the immutable version of redux-form reducer.

**Comment by jccf091** [comment]
tl;dr
look at the onclick logic for adding new field under your renderArrayFields function.
it should be like this:
```
<button type="button" onClick={() => fields.push(Immutable.Map())}>Add new member</button>
````
instead of:
```
<button type="button" onClick={() => fields.push({})}>Add new member</button>
````

When you try to type text into one of the field in FieldArray, `state.setIn(toPath(field), value)` will be called.
https://github.com/erikras/redux-form/blob/v6.0.0-rc.1/src/structure/immutable/index.js#L10

for example:
`
state.setIn(["values", "members", "0", "first_name"], 1)
`
If your are using fields.push({}), `Uncaught Error: invalid keyPath` error will popup.
because of: 
`
typeof state.getIn(['values', 'members', '0']) === "object"
`

**Comment by tdziezyk** [comment]
@jccf091 It is working after you changes.
Thank you all :)

**Comment by oakley808** [comment]
Seems like you'd also need to modify the way FieldArray removes fields 
https://github.com/erikras/redux-form/blob/a68d1a92d76ab361b1d916d437f17f5c16e63a9f/examples/fieldArrays/src/FieldArraysForm.js#L49-L53
```
<button
  type="button"
  title="Remove Hobby"
  onClick={() => fields.remove(index)}/>
```
I get an `VM362197:1 Uncaught TypeError: list.splice is not a function` error when I remove a field.

**Comment by marcosideame** [comment]
@oakley808, do you have the solution?

**Comment by oakley808** [comment]
@marcosideame Nope.

**Comment by mkchung** [comment]
@oakley808 
```
<button
  type="button"
  title="Remove Hobby"
  onClick={() => fields.remove(index)}/>
```

change to:

```
<button
  type="button"
  title="Remove Hobby"
  onClick={() => fields.remove(index + '')}/>
```

look at the "onClick" function, “index” convert to string
